libs/core/src/lib/date-picker/date-picker.component.ts

```diff
@@ -417,21 +420,35 @@ export class DatePickerComponent implements ControlValueAccessor, Validator {
             if (!CalendarService.datesEqual(firstDate, this.selectedRangeDate.start) ||
                 !CalendarService.datesEqual(secondDate, this.selectedRangeDate.end)) {
 
-                this.isInvalidDateInput = !firstDate.isDateValid() || !secondDate.isDateValid();
-
+                let selectedRangeDate: FdRangeDate = null;
 
                 /** If the end date is before the start date, there is need to replace them  */
                 if ((firstDate.getTimeStamp() > secondDate.getTimeStamp()) && secondDate.isDateValid()) {
-                    this.selectedRangeDate = { start: secondDate, end: firstDate };
+                    selectedRangeDate = { start: secondDate, end: firstDate };
                 } else {
-                    this.selectedRangeDate = { start: firstDate, end: secondDate };
+                    selectedRangeDate = { start: firstDate, end: secondDate };
+                }
+
+                this.isInvalidDateInput = !this.isRangeModelValid(selectedRangeDate);
+
+                /** If start date is invalid, cause of format, block or disable function, there is invalidDate forced */
+                if (!this.isStartDateValid(selectedRangeDate.start)) {
+                    selectedRangeDate.start = this.invalidDate();
+                }
+
+                /** If end date is invalid, cause of format, block or disable function, there is invalidDate forced */
```

Comment:
do you mean because "because of format, block..."?

Reply:
It's changed to `because`